pkg/structs/structs.go

```diff
@@ -272,27 +272,45 @@ const (
 	ForkAltair ForkVersion = iota
 	ForkBellatrix
 	ForkCapella
+	ForkUnknown
 )
 
+func (v ForkVersion) String() string {
+	switch v {
+	case ForkAltair:
+		return "altair"
+	case ForkBellatrix:
+		return "bellatrix"
+	case ForkCapella:
+		return "capella"
+	case ForkUnknown:
+		return "unknown"
+	default:
+		return ""
+	}
+}
+
 func (fs ForkState) IsCapella(slot Slot) bool {
-	return slot.Epoch() >= fs.CapellaEpoch
+	return fs.CapellaEpoch > 0 && slot.Epoch() >= fs.CapellaEpoch
 }
 
 func (fs ForkState) IsBellatrix(slot Slot) bool {
-	return slot.Epoch() >= fs.BellatrixEpoch && slot.Epoch() < fs.CapellaEpoch
+	return fs.BellatrixEpoch > 0 && slot.Epoch() >= fs.BellatrixEpoch && slot.Epoch() < fs.CapellaEpoch
 }
 
-func (fs ForkState) Version(epoch uint64) ForkVersion {
-
-	return ForkBellatrix // temoporary override
+func (fs ForkState) IsAltair(slot Slot) bool {
+	return slot.Epoch() >= fs.AltairEpoch && slot.Epoch() < fs.BellatrixEpoch
+}
 
-	switch {
-	case epoch >= uint64(fs.CapellaEpoch):
-		return ForkCapella
-	case epoch >= uint64(fs.BellatrixEpoch) && epoch < uint64(fs.CapellaEpoch):
-		return ForkBellatrix
-	case epoch < uint64(fs.BellatrixEpoch):
+func (fs ForkState) Version(slot Slot) ForkVersion {
+	if fs.IsAltair(slot) {
```

Comment:
I know it would be nitpick but - I think we will most likely see newest blocks here, so the if should be in reverse order :)